api/src/main/java/org/apache/commons/rdf/api/BlankNodeOrIRI.java

```diff
@@ -19,7 +19,7 @@
 
 /**
  * This interface represents the {@link RDFTerm}s that may be used in the
- * subject position of an RDF-1.1 {@link Triple}, including {@link BlankNode}
+ * subject position of an RDF-1.1 {@link Triple}, that is {@link BlankNode}
```

Comment:
"either" is stronger - meaning that we say they are distinct. (Which I think they are)

if you agree we can then also say in `RDFTerm` that it also must be a `IRI`, `BlankNode` or `Literal`.  Perhaps using _SHOULD_ so that evil people can still make their `SparqlVariable` if they so wish - knowing well that this would break various stuff.

In `RDFTerm` I added:

 * Instances of RDFTerm SHOULD be an {@link IRI}, {@link BlankNode} or
 * {@link Literal}.

and in `BlankNodeOrIRI`:

>  * Instances of BlankNodeOrIRI SHOULD be a {@link BlankNode} or an {@link IRI}.